mujoco_warp/_src/collision_gjk.py

```diff
@@ -1216,13 +1230,709 @@ def _epa(tolerance2: float, epa_iterations: int, pt: Polytope, geom1: Geom, geom
   # return from valid face
   if idx > -1:
     x1, x2 = _epa_witness(pt, idx)
-    return -wp.sqrt(pt.face_norm2[idx]), x1, x2
-  return 0.0, wp.vec3(), wp.vec3()
+    return -wp.sqrt(pt.face_norm2[idx]), x1, x2, idx
+  return 0.0, wp.vec3(), wp.vec3(), -1
+
+
+# return number (1, 2 or 3) of dimensions of a simplex; reorder vertices if necessary
+@wp.func
+def feature_dim(face: wp.vec3i, vert_index: wp.array(dtype=int), vert: wp.array(dtype=wp.vec3)):
```

Comment:
none of these functions have return type annotations!  not urgent for this PR but we should add some